src/ophyd_async/core/_signal.py

```diff
@@ -540,35 +601,36 @@ async def wait_for_value(
 
         wait_for_value(device.num_captured, lambda v: v > 45, timeout=1)
     """
-    if callable(match):
-        checker = _ValueChecker(match, match.__name__)  # type: ignore
+
+    if callable(match_value):
+        checker = _ValueChecker(match_value, match_value.__name__)  # type: ignore
     else:
-        checker = _ValueChecker(lambda v: v == match, repr(match))
+        checker = _ValueChecker(lambda v: v == match_value, repr(match_value))
     await checker.wait_for_value(signal, timeout)
 
 
 async def set_and_wait_for_other_value(
     set_signal: SignalW[T],
     set_value: T,
-    read_signal: SignalR[S],
-    read_value: S,
+    match_signal: SignalR[S],
+    match_value: S | Callable[[S], bool],
     timeout: float = DEFAULT_TIMEOUT,
     set_timeout: float | None = None,
```

Comment:
We decided to add `wait_for_set_completion = True` (or something like that) to both functions and always return the status, but then pass `False` from the areaDetector utility function